### PR TITLE
Raise error if Fortify enabled and user doesn't exist

### PR DESCRIPTION
**Pull request by hesham**

### CODE REVIEW

**Comment by hesham** [comment]
[ch2691]

File: spec/fortify_spec.rb [hunk 1]
```diff
@@ -68,74 +68,101 @@
...
+
+            project = Project.first
+            expect(project.can?(:destroy)).to eq true
+          end
+        end
       end
+    end
 
-      it 'does not allow unpermitted destroys' do
+    context "when user is not set" do
```
Comment (brianisme) [inline spec/fortify_spec.rb]: We should have a test case here to make sure `user.can?(:update)` also raises the error

Comment (hesham) [reply spec/fortify_spec.rb]: @brianisme do you think that's a better place to raise the error? in the `can?` method?

Comment (hesham) [reply spec/fortify_spec.rb]: We can do this:
```
def can?(action, field=nil)
  raise InvalidUserError.new("Fortify user not set") if !Fortify.user

  policy.can?(action, field)
end
```

Comment (hesham) [reply spec/fortify_spec.rb]: We can also raise it `policy.can?`.

Comment (brianisme) [reply spec/fortify_spec.rb]: `policy.can?` feels like a better idea. So that the logic is confined inside of policy base instead of ActiveRecord base.

Comment (hesham) [reply spec/fortify_spec.rb]: @brianisme good point. thanks!

**Comment by hesham** [comment]
@brianisme simplified things a bit. now we raise an error if the user tries to initialize a policy without `Fortify.user`. This will apply to both scoping and validations. Hope this makes sense.

**Pull request merged**